model_analyzer/result/result_manager.py

```diff
@@ -227,12 +234,15 @@ def _compile_measurement(self, measurement, gpu_table_key,
         """
 
         perf_config = measurement.perf_config()
-        model_name = perf_config['model-name']
+        model_name = perf_config.get_config_model().model_name()
```

Comment:
Instead of having the perf_config store a reference to the config model, we can get the temporary name from the RunConfig stored in the RunResult in `compile_results` [here](https://github.com/triton-inference-server/model_analyzer/blob/2c393d75f14b1ea24932b3928a8aa0ef66de438a/model_analyzer/result/result_manager.py#L205) by calling `next_best_result.get_run_config().model_name()`. We can add the `model_name` parameter to both `_compile_measurements` and `_compile_measurement`.

Reply:
Thanks for the detailed comment. I fixed this.